### PR TITLE
Enhancements for KubeLB

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds the following:
* Extend cluster health status with KubeLB health check
* Support for enforcing KubeLB at the datacenter level

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* Extend cluster health status with KubeLB health check
* Support for enforcing KubeLB at the datacenter level
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: cmd/seed-controller-manager/wrappers_ee.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func setupControllers(ctrlCtx *controllerContext) error {
 	}
 
 	if err := kubelbcontroller.Add(ctrlCtx.mgr, ctrlCtx.runOptions.workerCount, ctrlCtx.runOptions.workerName, ctrlCtx.runOptions.overwriteRegistry, ctrlCtx.seedGetter, ctrlCtx.clientProvider, ctrlCtx.log, ctrlCtx.versions); err != nil {
-		return fmt.Errorf("failed to create kubeLB controller: %w", err)
+		return fmt.Errorf("failed to create KubeLB controller: %w", err)
```
Comment (ahmedwaleedmalik) [inline cmd/seed-controller-manager/wrappers_ee.go:53]: @xrstf ☮️

File: pkg/defaulting/cluster.go [hunk 2]
```diff
@@ -139,6 +139,13 @@ func DefaultClusterSpec(ctx context.Context, spec *kubermaticv1.ClusterSpec, tem
 	// default cluster networking parameters
 	spec.ClusterNetwork = DefaultClusterNetwork(spec.ClusterNetwork, kubermaticv1.ProviderType(spec.Cloud.ProviderName), spec.ExposeStrategy)
 
+	// If KubeLB is enforced, enable it.
+	if datacenter.Spec.KubeLB != nil && datacenter.Spec.KubeLB.Enforced {
+		spec.KubeLB = &kubermaticv1.KubeLB{
+			Enabled: true,
+		}
```
Comment (xrstf) [inline pkg/defaulting/cluster.go]: I foresee bugs when the KubeLB struct is extended with more fields other than `Enabled`. Then this code will wipe everything else the user has set in the Cluster object. Maybe it's better to already write this a bit more carefully.

File: pkg/ee/kubelb/resources/seed-cluster/deployment.go [hunk 3]
```diff
@@ -104,18 +104,18 @@ func DeploymentReconciler(data kubeLBData) reconciling.NamedDeploymentReconciler
 func DeploymentReconcilerWithoutInitWrapper(data kubeLBData) reconciling.NamedDeploymentReconcilerFactory {
 	return func() (string, reconciling.DeploymentReconciler) {
 		return Name, func(dep *appsv1.Deployment) (*appsv1.Deployment, error) {
-			dep.Name = Name
-			dep.Labels = resources.BaseAppLabels(Name, nil)
+			dep.Name = resources.KubeLBDeploymentName
```
Comment (xrstf) [inline pkg/ee/kubelb/resources/seed-cluster/deployment.go]: The reconciling framework sets name the name and namespace. It is never necessary to manually set it. Also line 106 still says `Name`.

Comment (ahmedwaleedmalik) [reply pkg/ee/kubelb/resources/seed-cluster/deployment.go]: Line 106 is for the image name

File: pkg/defaulting/cluster.go (hunk 2 above)
Comment (ahmedwaleedmalik) [reply pkg/defaulting/cluster.go]: Thanks for this one, otherwise it would have been a rigorous debugging session for someone :D

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**